src/Microsoft.Framework.DependencyInjection/ServiceProvider.cs

```diff
@@ -49,17 +56,19 @@ internal ServiceProvider(ServiceProvider parent)
         /// <returns></returns>
         public object GetService(Type serviceType)
         {
-            var realizedService = _table.RealizedServices.GetOrAdd(serviceType, key =>
+            var realizedService = _table.RealizedServices.GetOrAdd(serviceType, _createServiceAccessor);
+            return realizedService.Invoke(this);
+        }
+
+        private Func<ServiceProvider, object> CreateServiceAccessor(Type serviceType)
+        {
+            var callSite = GetServiceCallSite(serviceType, new HashSet<Type>());
+            if (callSite != null)
             {
-                var callSite = GetServiceCallSite(key, new HashSet<Type>());
-                if (callSite != null)
-                {
-                    return RealizeService(_table, key, callSite);
-                }
-                return _ => null;
-            });
+                return RealizeService(_table, serviceType, callSite);
```

Comment:
Yes, but only once, then it's cached by `ServiceTable`. The problem described in the bug is one endemic to `GetOrAdd` - you usually have to allocate just to call the function.